lua/orgmode/utils.lua

```diff
@@ -370,4 +371,55 @@ function utils.prompt_autocomplete(arg_lead, list, split_chars)
   end, matches)
 end
 
+function utils.get_nearest_src_block(file, cursor)
```

Comment:
This function is very specific to the src blocks. Would a general function that finds the nearest node do the trick here?
For example, `block` node has `name`, `parameters` and `contents` children. If cursor is on the `block` node, just return it. If it's children, go up the parent until you find the required node. For example `utils.get_parents_until(node, 'block')` would return block node.

Reply:
There's already a function in `utils` that does something similar to this, however it automatically assumes that it should traverse from the parent of the passed node, rather than the passed node itself. I've just modified it to allow traversing from the passed node.

On top of that there's a simple wrapper for getting named child nodes (allows for filtering by name or grabbing all of them), and then a wrapper for getting a block node of a given name. That's about as simplified/abstract as I was able to get it, let me know if that looks good.

Reply:
Oh, you're right. I completely missed that. Looks ok.